### ISSUE TITLE
Column ZToken of the iOS app contains a 96 character token

### ISSUE THREAD

**Issue by syssi** [rytilahti/python-mirobo#75]
The ZToken column of the iOS app is encrypted now. The token cannot be used anymore by copy&paste because it has 96 characters now. The encryption is AES-ECB with an empty key. It can be deciphered by:

```python
from Crypto.Cipher import AES
import binascii
keystring = '00000000000000000000000000000000'
iostoken = '16a935db276e19b1c7bef0739f2deb7d3aca43c5e26a3c6445351cb2fb8495ea0143db63ee66b0cdff9f69917680151e'
key = bytes.fromhex(keystring)
cipher = AES.new(key, AES.MODE_ECB)
token = cipher.decrypt(bytes.fromhex(iostoken[:64]))
print(token)
# Output: b'4a447a41467058496746505541597033'
```

The support for encrypted tokens could be implemented now.

https://community.home-assistant.io/t/beta-xiaomi-gateway-integration/8213/3052
https://community.home-assistant.io/t/issue-config-xiaomi-vacuum-by-v0-51-1/24304/12
https://community.home-assistant.io/t/xiaomi-philips-light-token-problem/26701

**Comment by rytilahti** [comment]
I have not yet looked into how the extraction works on iOS, but I think we could provide a small external tool to do automatic extraction of usable tokens if that's too much effort. For Android this would require extracting the backup file (the hard part to do in python?) and reading the sqlite db for values (easily solvable).

**Comment by syssi** [comment]
This was my thought: Just implement the 10 lines above to support 32char and 96char tokens. Internally there is the 32char token used always. If the user supplies a 96char token it will be decrypted on the fly.

**Comment by rytilahti** [comment]
I used the code in that PR, but as I have no iDevices I can't really test it, so any test would be appreciated :-) (currently it will dump also raw information from the DB, as it was not clear what kind of info we can get from the DB.

**Comment by rytilahti** [comment]
This has been fixed with #77, so I think we can close this.
